### PR TITLE
Check if Indexer is listening on Port 9300 after install

### PR DESCRIPTION
**Pull request by c-bordon**
|Related issue|
|---|
|closes https://github.com/wazuh/wazuh-packages/issues/1411|

## Description

A validation is added so that before starting the Wazuh Indexer cluster it is verified that the nodes are listening on port 9300, a delay and a retry of 12 times are added to give time if it is due to some delay in the process of install

## Logs example

Example of an install where all nodes are listening on port 9300

Screenshot_20220406_124137

Example with a node with port 9300 blocked by firewall-cmd

Screenshot_20220406_125253
Screenshot_20220406_125138

## Tests

Centos: https://devel.ci.wazuh.info/view/Tests/job/Test_unattended/515/console
Ubuntu: https://devel.ci.wazuh.info/view/Tests/job/Test_unattended/516/console
Distributted: https://devel.ci.wazuh.info/view/Tests/job/Test_unattended_distributed/403/parameters/

### CODE REVIEW

File: unattended_installer/install_functions/installMain.sh [hunk 1]
```diff
@@ -272,7 +272,7 @@ function main() {
 
 # -------------- Start Elasticsearch cluster case  ------------------
```
Comment (rauldpm) [inline unattended_installer/install_functions/installMain.sh:273]: Change Elasticsearch name please

**Review by rauldpm — CHANGES_REQUESTED** [review]

File: unattended_installer/install_functions/indexer.sh [hunk 2]
```diff
@@ -154,6 +154,19 @@ function indexer_install() {
 
 function indexer_startCluster() {
 
+    f=0
```
[the remark this answers is not in the document]
Comment (c-bordon) [reply unattended_installer/install_functions/indexer.sh]: Done: https://github.com/wazuh/wazuh-packages/commit/d169f751a445ceddbfc0e4f28bb7e32f46f9783d

File: unattended_installer/install_functions/indexer.sh [hunk 3]
```diff
@@ -154,6 +154,19 @@ function indexer_install() {
 
 function indexer_startCluster() {
 
+    f=0    
+    for r in "${indexer_node_ips[@]}"; do
```
[the remark this answers is not in the document]
Comment (c-bordon) [reply unattended_installer/install_functions/indexer.sh]: Done: https://github.com/wazuh/wazuh-packages/commit/d169f751a445ceddbfc0e4f28bb7e32f46f9783d

File: unattended_installer/install_functions/installMain.sh (hunk 1 above)
Comment (c-bordon) [reply unattended_installer/install_functions/installMain.sh]: Done: https://github.com/wazuh/wazuh-packages/commit/d169f751a445ceddbfc0e4f28bb7e32f46f9783d

File: unattended_installer/install_functions/indexer.sh [hunk 4]
```diff
@@ -154,6 +154,19 @@ function indexer_install() {
...
 
+    retries=0    
+    for ip_to_test in "${indexer_node_ips[@]}"; do
+        until eval "nc -z ${ip_to_test} 9300" || [ "${retries}" -eq 12 ]; do
+            sleep 10
+            retries=$((retries+1))
+        done
+        if [ ${retries} -eq 12 ]; then
+            common_logger -e "Cannot initialize Wazuh indexer cluster. Unable to connect to node ${ip_to_test} on port 9300."
+            installCommon_rollBack
```
Comment (alberpilot) [inline unattended_installer/install_functions/indexer.sh:165]: Please avoid the rollBack in distributed mode.

File: unattended_installer/install_functions/indexer.sh [hunk 5]
```diff
@@ -154,6 +154,19 @@ function indexer_install() {
...
 function indexer_startCluster() {
 
+    retries=0    
+    for ip_to_test in "${indexer_node_ips[@]}"; do
+        until eval "nc -z ${ip_to_test} 9300" || [ "${retries}" -eq 12 ]; do
+            sleep 10
+            retries=$((retries+1))
+        done
+        if [ ${retries} -eq 12 ]; then
+            common_logger -e "Cannot initialize Wazuh indexer cluster. Unable to connect to node ${ip_to_test} on port 9300."
```
Comment (alberpilot) [inline unattended_installer/install_functions/indexer.sh:164]:
```suggestion
            common_logger -e "Cannot initialize Wazuh indexer cluster. Unable to connect to node ${ip_to_test} on port 9300. Review your firewall/network configuration before and try again."
```

**Review by alberpilot — CHANGES_REQUESTED** [review]

File: unattended_installer/install_functions/indexer.sh (hunk 4 above)
Comment (c-bordon) [reply unattended_installer/install_functions/indexer.sh]: Done: https://github.com/wazuh/wazuh-packages/commit/e5962cd7f8400b1e89f4210190215cb3b6c09c05

**Review by okynos — APPROVED** [review]
LGTM

**Review by alberpilot — APPROVED** [review]
LGTM

**Pull request merged**